molo/core/content_import/helpers/summaries.py

```diff
@@ -0,0 +1,27 @@
+import requests
+
+from molo.core.content_import.errors import SiteResponseError
+
+
+def get_summaries(url):
+    # TODO handle error responses
+    # TODO validate json response
+    resp = requests.get(url)
+
+    if is_error_response(resp):
+        raise SiteResponseError(
+            "%s responded with a %s error" % (url, resp.status_code))
+    else:
+        return parse_repo_summaries_response(resp.json())
+
+
+def parse_repo_summaries_response(resp):
+    return [{
+        'name': d['index'],
+        'title': d['data']['title']
+    } for d in resp]
+
+
+def is_error_response(resp):
+    code = resp.status_code
+    return 400 <= code < 500 or 500 <= code < 600
```

Comment:
> can't this be simplified to return 400 <= code < 600

No idea why I didn't just do that :)

> And wouldn't a more explicit return code in [200, 201, 204, 301, 302] be easier to manage?

I can't think of any examples of when other status codes would be used here, but an error range feels more robust.